pkg/cloud/scope/clients.go

```diff
@@ -139,15 +141,27 @@ func NewSTSClient(scopeUser cloud.ScopeUsage, session cloud.Session, target runt
 }
 
 // NewSSMClient creates a new Secrets API client for a given session
-func NewSSMClient(scopeUser cloud.ScopeUsage, session cloud.Session, target runtime.Object) ssmiface.SSMAPI {
-	ssmClient := ssm.New(session.Session())
+func NewSSMClient(scopeUser cloud.ScopeUsage, session cloud.Session, logger logr.Logger, target runtime.Object) ssmiface.SSMAPI {
+	ssmClient := ssm.New(session.Session(), aws.NewConfig().WithLogLevel(getAWSLogLevel(logger)))
 	ssmClient.Handlers.Build.PushFrontNamed(getUserAgentHandler())
 	ssmClient.Handlers.CompleteAttempt.PushFront(awsmetrics.CaptureRequestMetrics(scopeUser.ControllerName()))
 	ssmClient.Handlers.Complete.PushBack(recordAWSPermissionsIssue(target))
 
 	return ssmClient
 }
 
+func getAWSLogLevel(logger logr.Logger) aws.LogLevelType {
+	if logger.V(4).Enabled() {
```

Comment:
I think we'd probably want the log levels at which the AWS logging is enabled to be higher. This is so the level is differentiated from normal CAPA debugging.

Perhaps `9` for `aws.LogDebug` and `10` for `aws.LogDebugWithHTTPBody`....so something similar.

It would be good to define the log levels as constants somewhere so that they can be used and throughout the rest of the provider (we generally use levels 2 and 4 already)